### DISCUSSION TITLE
Feature request: return all output files from `quarto inspect`

### DISCUSSION THREAD

**Discussion by wlandau** [quarto-dev/quarto-cli#2378] [Feature Requests]
The [`targets`](https://docs.ropensci.org) R package [integrates with Quarto](https://books.ropensci.org/targets/literate-programming.html#quarto-targets) via `tarchetypes::tar_quarto()` for reproducible analysis pipelines. For reproducibility, `targets` hashes the input and output files from Quarto and uses the hashes to decide whether to run the Quarto document or project. To learn which input and output files to  track, `tar_quarto()` digs through the output of `quarto inspect` and tries to figure out which input files Quarto needs and which output files Quarto will generate. It mostly works, but there are edge cases like https://github.com/ropensci/tarchetypes/discussions/102 and https://github.com/ropensci/tarchetypes/discussions/107 (mentioned in https://github.com/quarto-dev/quarto-r/issues/14) (and it is [not blazing fast](https://github.com/quarto-dev/quarto-cli/discussions/1266) either). On the other hand, `tarchetypes::tar_quarto()` is not in a position to circumvent `quarto inspect` and manually emulate the internal decisions and behaviors of Quarto. So it would be extremely helpful if Quarto  would straightforwardly list all its input and output file paths for a project or document.

**Comment by cderv** [comment]
@allenmanning @dragonstyle We discuss `quarto inspect` usage at our last meeting. This is an example of usage for the command in an integration context with an R package. 

Also this Idea posted on Discussion in September may have been lost in triage: 
@dragonstyle do `quarto inspect` already return all output files ? Is this something in the scope ?

**Reply by cderv** [reply]
I opened https://github.com/quarto-dev/quarto-cli/issues/8397 to track this in our issue board directly
